src/features/web-compat.js

```diff
@@ -635,10 +643,25 @@ export default class WebCompat extends ContentFeature {
                 // Chromium accepts float values for initial-scale
                 const parsedInitialScale = parseFloat(initialScalePart[1])
                 if (parsedInitialScale !== 1) {
-                    this.forceViewportTag(viewportTag, `width=device-width, ${viewportContent}`)
+                    forcedValues.width = 'device-width'
                 }
             }
         }
+
+        const newContent = []
+        'width' in forcedValues && newContent.push(`width=${forcedValues.width}`)
+        'initial-scale' in forcedValues && newContent.push(`initial-scale=${forcedValues['initial-scale']}`)
+        'user-scalable' in forcedValues && newContent.push(`user-scalable=${forcedValues['user-scalable']}`)
+        'maximum-scale' in forcedValues && newContent.push(`maximum-scale=${forcedValues['maximum-scale']}`)
+        if (newContent.length > 0) { // need to override at least one viewport component
+            parsedViewportContent.forEach(([key], idx) => {
+                // @ts-expect-error hasOwn does exist
+                if (!(Object.hasOwn(forcedValues, key))) {
+                    newContent.push(`${viewportContentParts[idx]}`) // reuse the original values, not the parsed ones
```

Comment:
```suggestion
                    newContent.push(viewportContentParts[idx]) // reuse the original values, not the parsed ones
```

Maybe run `trim()` here if you wanted to remove the whitespace rather than making the regex more complex?